docs/rst/knora-api-server/api_v2/knora-iris.rst

```diff
@@ -18,7 +18,96 @@
+
+::
+
+   http://HOST[:PORT]/ontology/ONTOLOGY_NAME/API_VERSION
+
+For built-in ontologies, the host is always ``api.knora.org``. Otherwise, the hostname and port configured in ``application.conf`` under ``app.http.knora-api.host`` and ``app.http.knora-api.http-port`` are used (the port is omitted if it is 80). This means that when a built-in external ontology IRI is dereferenced, the ontology can be served by a Knora API server running at ``api.knora.org``. When a project-specific external ontology IRI is dereferenced, the ontology can be served by the Knora API server that hosts the project. During development and testing, this could be ``localhost``.
+
+The name of an external ontology is the same as the name of the corresponding internal ontology, with one exception: the external form of ``knora-base`` is called ``knora-api``.
+
+The API version identifier indicates not only the version of the API, but also an API 'schema'. The Knora API v2 is available in two schemas:
+
+- A default schema, which is suitable both for reading and for editing data. The default schema represents values primarily as complex objects. Its version identifier is ``v2``.
+- A simple schema, which is suitable for reading data but not for editing it. The simple schema facilitates interoperability between Knora ontologies and non-Knora ontologies, since it represents values primarily as literals. Its version identifier is ``simple/v2``.
+
+Other schemas could be added in the future for more specific use cases.
+
+When requesting an ontology, the client requests a particular schema. (This will also be true of all Knora API v2 requests: the client will be able to specify which schema the response should be provided in.)
+
+For example, suppose a Knora API server is running at ``knora.example.org`` and hosts the ``incunabula`` ontology. That ontology can then be requested using either of these IRIs:
+
+- ``http://knora.example.org/ontology/incunabula/v2`` (for the default schema)
+- ``http://knora.example.org/ontology/incunabula/simple/v2`` (for the simple schema)
+
+While the internal ``incunabula`` ontology refers to definitions in ``knora-base``, the external ``incunabula`` ontology that is served by the API refers instead to a ``knora-api`` ontology, whose IRI depends on the schema being used:
+
+- ``http://api.knora.org/ontology/knora-api/v2`` (for the default schema)
+- ``http://api.knora.org/ontology/knora-api/simple/v2`` (for the simple schema)
+
+Ontology Entity IRIs
+^^^^^^^^^^^^^^^^^^^^
+
+Knora ontologies use 'hash namespaces' (see `URI Namespaces`_). This means that the IRI of an ontology entity (a class or property definition) is constructed by adding a hash character (``#``) to the ontology IRI, followed by the name of the entity. In Knora, an entity name must be a valid XML NCName_. Thus, the class ``incunabula:book`` has the following IRIs:
+
+- ``http://www.knora.org/ontology/incunabula#book`` (for the internal entity)
```

Comment:
Shouldn't we also use projectUUID in these IRIs? Because if we don't, chances are that we will have to face duplicate in the names of the project with the (hopefully) growing number of DaSCH satellites.

But we should keep the idea of the `knora-base:projectShortName` property to be used by the GUI as discussed in #595 and add a `knora-base:projectCode`corresponding the the projectUUID.

Reply:
By `projectUUID`, I actually meant `projectcode`, as until now we haven't mentionned projectUUID (at least I didn't understand it that way).

>An ontology can exist independently of a project. There will be ontologies shared by multiple projects. So I don’t think we can include any sort of project ID (whether UUID, shortname, or project code) in an ontology IRI. 

Yes, but if you are talking about the kind of standard ontologies shared by various projects, my guess is that these ontologies will undergo a review process. Their status will also change as they proceed from 'project-specific ontologies' to 'standard built-in ontologies'. In the process, aside from cleaning and improving these ontology, can't we imagine that the initial project-ID attached to the IRIs will be changed into a more generic name (such as bibliography, for example)?

Reply:
> I could imagine a problem occurring if someday we do federated searches in which triplestores on different servers are talking to each other.

I don't know if we are ever going to support federated searches on the triplestore level. More likely on the federated `webapi` server level. In those cases, the different hostnames will give us unique names.

>  Or if two ontologies created with the same name on different servers are later moved to the same server.

Maybe it would be easier to require a name change in those rare cases?

Reply:
@mrivoal A UUID is not something assigned manually, it’s a very long random number, which the Knora API server encodes as a 22-character string of letters and digits.

The problem with changing the IRI of an ontology is that you also have to change all the data that uses the ontology. In the case of a project-specific ontology that became standardised, I think ideally there would be a project-specific ‘bridge’ ontology so the data wouldn’t have to change.

So... maybe including a short project ID (not a UUID) in the ontology IRI wouldn’t be such a bad idea. But it would mean that all existing ontologies and data would have to be updated.

Reply:
> @mrivoal A UUID is not something assigned manually, it’s a very long random number, which the Knora API server encodes as a 22-character string of letters and digits.

Yes, I figured it out :)

> The problem with changing the IRI of an ontology is that you also have to change all the data that uses the ontology. 
I am very aware of that, we have already faced the problem when we changed the syntax of  our IRIs to include the projectcode instead of the projetshortname.

> So... maybe including a short project ID (not a UUID) in the ontology IRI wouldn’t be such a bad idea. But it would mean that all existing ontologies and data would have to be updated.

Honestly, I thought this was the road we all agreed on several months ago.

Reply:
> Honestly, I thought this was the road we all agreed on several months ago.

I thought we had only agreed that project IDs would be used in resource IRIs. I don’t remember anything about including them in ontology IRIs. But I’m not known for having the best memory. Anyway, we can certainly do it, but it will take some work to convert everything.

Reply:
We decided that there would be project ID, and that IRIs would be `http://rdfh.ch/PROJECT_ID`, we didn't specify clearly "resource" IRI and "project" IRI (c.f. minutes by mail from Lukas dated 2017-03-16).  
So we apparently left room for misunderstandings.  
As this was to avoid collisions, we thought it should be in the ontology too.

Reply:
I’d like to see if I can make this work in a backwards-compatible way, so existing data doesn’t have to be changed.